embedded-graphics/src/primitives/mod.rs

```diff
@@ -47,6 +66,13 @@ macro_rules! egcircle {
         $crate::primitives::Circle::new($crate::geometry::Point::new($cx, $cy), $r)
             $( .$style_key($style_value) )*
     }};
+
+    ($center:expr, $r:expr $(, $style_key:ident = $style_value:expr )* $(,)?) => {{
```

Comment:
Wouldn't it be possible to use `Point::from($center)` in the macros? This should work with all types that implement `From<...> for Point` (including tuples and `Point` expressions) and only requires one macro with `$center::tt` as first parameter.